File: OnlinePK-Android/biz-live/src/main/java/com/netease/biz_live/yunxin/live/dialog/AudienceConnectDialog.java
package com.netease.biz_live.yunxin.live.dialog;

import android.graphics.drawable.GradientDrawable;
import android.os.Bundle;
import android.view.Gravity;
import android.view.View;
import android.view.ViewGroup;
import android.view.Window;
import android.view.WindowManager;
import android.widget.TextView;

import androidx.annotation.Nullable;
import androidx.viewpager.widget.ViewPager;

import com.blankj.utilcode.util.ColorUtils;
import com.blankj.utilcode.util.ScreenUtils;
import com.blankj.utilcode.util.ViewUtils;
import com.google.android.material.tabs.TabLayout;
import com.netease.biz_live.R;
import com.netease.biz_live.yunxin.live.dialog.adapter.AudiencePageAdapter;
import com.netease.yunxin.nertc.demo.utils.SpUtils;


/**
 * 观众连麦dialog
 */
public class AudienceConnectDialog extends BaseBottomDialog{

    private TextView tvInviteCount;

    private TextView tvApplyCount;

    private TextView tvConnectManage;

    private String roomId;

    public static final String ROOM_ID = "roomId";

    public AudienceConnectDialog(){

    }

    @Override
    protected int getResourceLayout() {
        return R.layout.audience_connect_dialog_layout;
    }

    @Override
    protected void initView(View rootView) {
        super.initView(rootView);

        if(getArguments() != null){
            Bundle bundle = getArguments();
            roomId = bundle.getString(ROOM_ID);
        }

        ViewPager audiencePages = rootView.findViewById(R.id.vp_audience);
        audiencePages.setAdapter(new AudiencePageAdapter(getChildFragmentManager(),roomId));
        audiencePages.setOffscreenPageLimit(3);

        TabLayout tabLayout = rootView.findViewById(R.id.tab_audience_type);
        tabLayout.setupWithViewPager(audiencePages);
        tabLayout.removeAllTabs();
        tabLayout.setTabGravity(TabLayout.GRAVITY_FILL);
        TabLayout.Tab tab1 = tabLayout.newTab().setCustomView(R.layout.view_item_audience_tab);
        tvInviteCount = tab1.getCustomView().findViewById(R.id.tv_tab_name);
        tvInviteCount.setText("邀请上麦");
        tabLayout.addTab(tab1, 0, false);
        TabLayout.Tab tab2 = tabLayout.newTab().setCustomView(R.layout.view_item_audience_tab);
        tvApplyCount = tab2.getCustomView().findViewById(R.id.tv_tab_name);
        tvApplyCount.setText("连麦申请");
        tabLayout.addTab(tab2, 1, true);
        TabLayout.Tab tab3 = tabLayout.newTab().setCustomView(R.layout.view_item_audience_tab);
        tvConnectManage = tab3.getCustomView().findViewById(R.id.tv_tab_name);
        tvConnectManage.setText("连麦管理");
        tabLayout.addTab(tab3, 2, false);
        audiencePages.setCurrentItem(1);
    }

    @Override
    protected void initParams() {
        Window window = getDialog().getWindow();
        if (window != null) {
            window.setBackgroundDrawableResource(R.drawable.white_corner_bottom_dialog_bg);

            WindowManager.LayoutParams params = window.getAttributes();
            params.gravity = Gravity.BOTTOM;
            // 使用ViewGroup.LayoutParams，以便Dialog 宽度充满整个屏幕
            params.width = ViewGroup.LayoutParams.MATCH_PARENT;
            params.height = ScreenUtils.getScreenHeight() / 3;
            window.setAttributes(params);

        }
        setCancelable(true);//设置点击外部是否消失
    }


}

File: OnlinePK-Android/biz-live/src/main/java/com/netease/biz_live/yunxin/live/anchor/dialog/AudienceConnectDialog.kt
/*
 *  Copyright (c) 2021 NetEase, Inc.  All rights reserved.
 *  Use of this source code is governed by a MIT license that can be found in the LICENSE file
 */

package com.netease.biz_live.yunxin.live.anchor.dialog

import android.view.Gravity
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.viewpager.widget.ViewPager
import com.blankj.utilcode.util.ScreenUtils
import com.google.android.material.tabs.TabLayout
import com.netease.biz_live.R
import com.netease.biz_live.yunxin.live.anchor.dialog.adapter.AudiencePageAdapter
import com.netease.biz_live.yunxin.live.dialog.BaseBottomDialog

/**
 * 观众连麦dialog
 */
class AudienceConnectDialog : BaseBottomDialog() {
    private var tvInviteCount: TextView? = null
    private var tvApplyCount: TextView? = null
    private var tvConnectManage: TextView? = null
    private var roomId: String? = null
    override fun getResourceLayout(): Int {
        return R.layout.audience_connect_dialog_layout
    }

    override fun initView(rootView: View) {
        super.initView(rootView)
        arguments?.let {
            val bundle = it
            roomId = bundle.getString(ROOM_ID)
        }
        val audiencePages: ViewPager = rootView.findViewById(R.id.vp_audience)
        audiencePages.adapter = roomId?.let { AudiencePageAdapter(childFragmentManager, it) }
        audiencePages.offscreenPageLimit = 3
        val tabLayout: TabLayout = rootView.findViewById(R.id.tab_audience_type)
        tabLayout.setupWithViewPager(audiencePages)
        tabLayout.removeAllTabs()
        tabLayout.tabGravity = TabLayout.GRAVITY_FILL
        val tab1 = tabLayout.newTab().setCustomView(R.layout.view_item_audience_tab)
        tvInviteCount = tab1.customView?.findViewById(R.id.tv_tab_name)
        tvInviteCount?.text = getString(R.string.biz_live_invite_join_seats)
        tabLayout.addTab(tab1, 0, false)
        val tab2 = tabLayout.newTab().setCustomView(R.layout.view_item_audience_tab)
        tvApplyCount = tab2.customView?.findViewById(R.id.tv_tab_name)
        tvApplyCount?.setText(R.string.biz_live_apply_seat)
        tabLayout.addTab(tab2, 1, true)
        val tab3 = tabLayout.newTab().setCustomView(R.layout.view_item_audience_tab)
        tvConnectManage = tab3.customView?.findViewById(R.id.tv_tab_name)
        tvConnectManage?.setText(R.string.biz_live_apply_seat_manager)
        tabLayout.addTab(tab3, 2, false)
        audiencePages.currentItem = 1
    }

    override fun initParams() {
        val window = dialog?.window
        window?.let {
            it.setBackgroundDrawableResource(R.drawable.white_corner_bottom_dialog_bg)
            val params = it.attributes
            params.gravity = Gravity.BOTTOM
            // 使用ViewGroup.LayoutParams，以便Dialog 宽度充满整个屏幕
            params.width = ViewGroup.LayoutParams.MATCH_PARENT
            params.height = ScreenUtils.getScreenHeight() / 3
            it.attributes = params
        }
        isCancelable = true //设置点击外部是否消失
    }

    companion object {
        const val ROOM_ID: String = "roomId"
    }
}